database/assertion.go

```diff
@@ -0,0 +1,50 @@
+package database
+
+import (
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/ethdb"
+	"github.com/ethereum/go-ethereum/core/rawdb"
+	"github.com/ethereum/go-ethereum/core/state"
+	
+	"github.com/lightstreams-network/lightchain/tracer"
+)
+
+func assertPersistedGenesisBlock(t *tracer.Tracer, genesis core.Genesis, chainDataDir string) {
```

Comment:
you could have used the `git mv` command @ggarri. Also, why did you remove the nullable tracer? Now we have to use a flag all over the codebase every time we want to do an assertion which is already 5 times. I used the Nullable object pattern to solve this if (trc.Enable) {... flag issue. 

We didn't discuss this refactor also it will cause me a lot of conflicts now. I think this is not an improvement. 

Could u not remove the code just like that? U could still have a global tracer that has common logic between db/consensus and configurations (as we dicussed) and inject them to the object u removed without removing the whole  EthDBTracer and Nullable tracer.

Reply:
In this case I cannot really see the need of having  NullTracer because my intention is to remove the entire need of a tracer per module and have a single one which is used globally. I think that if/else you mention won't be so repetitive and it is worthy in order to reduce maintaining 2 parallel struct only for that propose. 

Lets have a chat p2p on it

Reply:
ok, let's discuss IRL

Reply:
IRL UPDATE: The refactor was reverted back to the current master impl with NullTracer to avoid programming with flags